lib/contentful/sys_data.ex

```diff
@@ -1,15 +1,23 @@
 defmodule Contentful.SysData do
+  alias Contentful.ContentType
+
   @moduledoc """
   The SysData represents internal additional data for Contetnful API objects, usually found in the
   "sys" part of the response objects. It's also referred to as "common properties".
 
   See the [official documentation for more information](https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/locales).
   """
-  defstruct [:id, :revision, :version]
+  defstruct [:id, :revision, :version, :created_at, :updated_at, locale: nil, content_type: nil]
 
   @type t :: %Contentful.SysData{
           id: String.t(),
           revision: integer() | nil,
-          version: integer() | nil
+          version: integer() | nil,
+          created_at: String.t() | nil,
+          updated_at: String.t() | nil,
+          # NOTE: locale string only exists in entries and assets
+          locale: String | nil,
+          # NOTE: ContentType only exists for entries
+          content_type: %ContentType{} | nil
```

Comment:
Here I wish we had some kind of dependent typing (if the thing is an entry the content type must present).
A less pragmatic way would be to have a sys data enum with specific variants for the different data types; I'm currently experimenting with that idea in a hobby SDK.

Reply:
I am not completely sure, if we really should wrap this into a `ContentType` here. Ultimately I opted for consistency, since it'll give you a possibility to match this at run time.

The `ContentType` isn't complete though, since it gets manually created from a `"Link"`.

I am still looking into how to generically allow for resolving links from arbitrary entities.

Reply:
I'd actually revert this to just be an optional `String.t()` I guess